spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/watcher/DefaultDependencyWatcher.java

```diff
@@ -0,0 +1,69 @@
+package org.springframework.cloud.zookeeper.discovery.watcher;
+
+import org.apache.curator.x.discovery.ServiceCache;
+import org.apache.curator.x.discovery.ServiceDiscovery;
+import org.springframework.cloud.zookeeper.discovery.ZookeeperDiscoveryClient;
+import org.springframework.cloud.zookeeper.discovery.watcher.presence.DependencyPresenceOnStartupVerifier;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultDependencyWatcher implements DependencyWatcher {
+
+    private final ServiceDiscovery serviceDiscovery;
+    private final Map<String, ServiceCache> dependencyRegistry = new HashMap<>();
+    private final List<DependencyWatcherListener> listeners = new ArrayList<>();
+    private final DependencyPresenceOnStartupVerifier dependencyPresenceOnStartupVerifier;
+    private final ZookeeperDiscoveryClient zookeeperDiscoveryClient;
+
+    public DefaultDependencyWatcher(ServiceDiscovery serviceDiscovery,
+                                    DependencyPresenceOnStartupVerifier dependencyPresenceOnStartupVerifier,
+                                    ZookeeperDiscoveryClient zookeeperDiscoveryClient) {
+        this.serviceDiscovery = serviceDiscovery;
+        this.dependencyPresenceOnStartupVerifier = dependencyPresenceOnStartupVerifier;
+        this.zookeeperDiscoveryClient = zookeeperDiscoveryClient;
+    }
+
+    @Override
+    public void registerDependencyRegistrationHooks() throws Exception {
+        for (String dependencyPath : getListOfServiceDependencies(zookeeperDiscoveryClient)) {
+            ServiceCache serviceCache = serviceDiscovery.serviceCacheBuilder().name(dependencyPath).build();
+            serviceCache.start();
+            dependencyPresenceOnStartupVerifier.verifyDependencyPresence(dependencyPath, serviceCache, retrieveFromMetadataIfServiceIsRequired(dependencyPath));
+            dependencyRegistry.put(dependencyPath, serviceCache);
+            serviceCache.addListener(new DependencyStateChangeListenerRegistry(listeners, dependencyPath, serviceCache));
+        }
+    }
+
+    private List<String> getListOfServiceDependencies(ZookeeperDiscoveryClient zookeeperDiscoveryClient) {
```

Comment:
Right now we have a microservice descriptor file that we want to migrate to properties:

https://github.com/4finance/boot-microservice/blob/master/src/main/resources/microservice.json

```
{
    "pl": {
        "this": "com/ofg/twitter-places-analyzer",
        "dependencies": {
            "collerator": {
                "path": "com/ofg/twitter-places-collerator"
            }
        }
    }
}
```

`pl` is the realm
`this` is in fact `spring.application.name`
in the `dependencies` section we provide
- alias `collerator` that we will then use in the code (that could be passed to ribbon)
- `path` - for the current alias path under given realm for a dependency

also what can be passed in this descriptor is:

https://github.com/4finance/micro-infra-spring/tree/master/micro-deps/micro-deps#microservicejson-file

```
 {
        "prod": {
            "this": "foo/bar/registration",
            "dependencies": {
                "users": "foo/bar/users",
                "newsletter": {
                    "path": "foo/bar/comms/newsletter",
                    "load-balancer": "random",
                    "contentTypeTemplate": "application/vnd.newsletter.$version+json",
                    "version": "v1"                    
                },
                "confirmation": {
                    "path": "foo/bar/security/confirmation",
                    "headers": {
                        "header1": "value1",
                        "header2": "value2"
                    }
                }
            }
        }
    }
```